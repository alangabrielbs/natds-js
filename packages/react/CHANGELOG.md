# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.5.4](https://github.com/natura-cosmeticos/natds-js/compare/v1.5.3...v1.5.4) (2021-03-16)

**Note:** Version bump only for package @naturacosmeticos/natds-react
