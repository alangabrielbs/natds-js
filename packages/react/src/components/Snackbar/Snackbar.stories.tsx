import React, { useRef, useState } from 'react'
import { Story, Meta } from '@storybook/react'
import { Snackbar, SnackbarProps } from '.'
import { refProp } from './Snackbar'
import Button from '../Button'
import Icon from '../Icon'

const componentStatus = `

> Snackbar component is used for instant notifications

## Properties
| Property              | Values                                                                      |    Status       |
|---                    |                                                                          ---|              ---|
| **variant**           | Standard                                                                    | ✅ Available    |
| **Title**             | any string                                                                  | ✅ Available    |
| **Icon**              | Component Icon                                                              | ✅ Available    |
| **directionButton**   | none, inlineButton, blockButton                                             | ✅ Available    |
| **Feedback**          | default, success, error, warning, info,                                     | ✅ Available    |
| **Position**          | topLeft, topCenter, topRight, <br />bottomLeft, bottomCenter, bottomRight   | ✅ Available    |
| **Timer**             | number                                                                      | ✅ Available    |
| **Animation**         | true/false                                                                  | ✅ Available    |
| **buttonComponent**   | Component Button                                                            | ✅ Available    |

## Technical Usages Examples
`

export default {
  title: 'Components/Snackbar',
  component: Snackbar,
  parameters: {
    componentSubtitle: '',
    docs: { description: { component: componentStatus } }
  }
} as Meta

export const Playground: Story<SnackbarProps> = (args) => {
  const [btnStatusD, setBtnStatus] = useState(false)
  const snackRef = useRef<refProp>(null)
  const btnStatus = () => {
    setBtnStatus(true)
    setTimeout(() => {
      setBtnStatus(false)
    }, 2000)
  }
  return (
    <>
      <Snackbar
        {...args}
        ref={snackRef}
      >
        Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quae veritat!
      </Snackbar>
      <Button
        disabled={btnStatusD}
        onClick={() => {
          btnStatus()
          return snackRef.current ? snackRef.current.show() : ''
        }}
      >
        Click
      </Button>
    </>
  )
}
export const WithIcon: Story<SnackbarProps> = (args) => {
  const [btnStatusD, setBtnStatus] = useState(false)
  const snackRef = useRef<refProp>(null)
  const btnStatus = () => {
    setBtnStatus(true)
    setTimeout(() => {
      setBtnStatus(false)
    }, 2000)
  }
  return (
    <>
      <Snackbar
        {...args}
        ref={snackRef}
        icon={
          // eslint-disable-next-line react/destructuring-assignment
          <Icon name="outlined-default-mockup" color={args.feedback === 'warning' ? 'highlight' : 'surface'} />
      }
      >
        Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quae veritat!
      </Snackbar>
      <Button
        disabled={btnStatusD}
        onClick={() => {
          btnStatus()
          return snackRef.current ? snackRef.current.show() : ''
        }}
      >
        Click
      </Button>
    </>
  )
}

export const WithButton: Story<SnackbarProps> = (args) => {
  const [btnStatusD, setBtnStatus] = useState(false)
  const snackRef = useRef<refProp>(null)
  const btnStatus = () => {
    setBtnStatus(true)
    setTimeout(() => {
      setBtnStatus(false)
    }, 2000)
  }
  return (
    <>
      <Snackbar
        {...args}
        ref={snackRef}
        buttonComponent={(
          <Button
            onClick={() => ''}
            variant="contained"
          >
            <span>Default</span>
          </Button>
        )}
      //   icon={
      //     <Icon name="outlined-default-mockup" color="surface" />
      // }
      >
        Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quae veritat!
      </Snackbar>
      <Button
        disabled={btnStatusD}
        onClick={() => {
          btnStatus()
          return snackRef.current ? snackRef.current.show() : ''
        }}
      >
        Click
      </Button>
    </>
  )
}
export const WithIconButton: Story<SnackbarProps> = (args) => {
  const [btnStatusD, setBtnStatus] = useState(false)
  const snackRef = useRef<refProp>(null)
  const btnStatus = () => {
    setBtnStatus(true)
    setTimeout(() => {
      setBtnStatus(false)
    }, 2000)
  }
  return (
    <>
      <Snackbar
        {...args}
        ref={snackRef}
        buttonComponent={(
          // eslint-disable-next-line react/destructuring-assignment
          <Icon name="outlined-default-mockup" color={args.feedback === 'warning' ? 'highlight' : 'surface'} />
        )}
      >
        Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quae veritat!
      </Snackbar>
      <Button
        disabled={btnStatusD}
        onClick={() => {
          btnStatus()
          return snackRef.current ? snackRef.current.show() : ''
        }}
      >
        Click
      </Button>
    </>
  )
}
