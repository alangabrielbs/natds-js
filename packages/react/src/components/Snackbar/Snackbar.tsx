import React, { useImperativeHandle, useState } from 'react'
import styles from './Snackbar.styles'
import { SnackbarProps } from './Snackbar.props'

export type refProp = {
  show: () => void;
};

const Snackbar = React.forwardRef<refProp, SnackbarProps>(
  // eslint-disable-next-line complexity
  (props, ref) => {
    const {
      className = '',
      position = 'topCenter',
      feedback = 'default',
      actionButton,
      timer,
      buttonComponent,
      ...rest
    } = props

    const {
      snackbarContainer, wrapper, wrapperRow, wrapperColumm, wrapperAction, bodyText,
      wrapperIcon, show, close
    } = styles({
      position, feedback, actionButton, timer
    })

    const [state, setState] = useState(false)
    useImperativeHandle(ref, () => ({
      show() {
        setState(!state)
        if (timer) {
          setTimeout(() => {
            setState(false)
          }, timer * 1000)
        }
      }
    }))

    return (
      <div
        className={`${className} ${state ? show : close} ${snackbarContainer}`}
        {...rest}
      >
        {props.children}
        <div className={wrapper}>
          <div className={wrapperRow}>
            {
              rest.icon && (
              <div className={wrapperIcon}>
                {rest.icon}
              </div>
              )
            }
            <div className={wrapperColumm}>
              {
                rest.title && <div className={bodyText}>{rest.title}</div>
              }
              <div className={bodyText}>
                Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quae veritat!
              </div>

            </div>
          </div>
          {
            actionButton !== 'none'
            && (
              <div className={wrapperAction}>

                <span>{buttonComponent}</span>

              </div>
            )
}
        </div>
      </div>
    )
  }
)

export default Snackbar
