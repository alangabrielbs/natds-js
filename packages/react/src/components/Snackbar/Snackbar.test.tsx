import React from 'react'
import { fireEvent, screen } from '@testing-library/react'
import Button from '../Button'
import Snackbar from './Snackbar'
import renderWithTheme from '../../helpers/renderWithTheme'

describe('Snackbar component', () => {
  it('should render correctly with default props', () => {
    const { styles, component } = renderWithTheme(<Snackbar timer={5} />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render title', () => {
    renderWithTheme(<Snackbar timer={5} title="Title" />)
    expect(screen.getByText('Title')).toBeInTheDocument()
    expect(screen.getByTestId('title-item')).toBeInTheDocument()
  })
  it('should render children', () => {
    renderWithTheme(<Snackbar timer={5}> Texto children</Snackbar>)
    expect(screen.getByText('Texto children')).toBeInTheDocument()
  })
  it('should render with clicked in ref.current.show()', () => {
    const onClickMock = jest.fn()
    const snackRef = {
      current: {
        show: onClickMock
      }
    }
    const { component: { getByTestId } } = renderWithTheme(
      <>
        <Button testID="btn-click" onClick={() => snackRef.current?.show()}>Click</Button>
        <Snackbar ref={snackRef} timer={5}> Texto children</Snackbar>
      </>
    )
    fireEvent.click(getByTestId('btn-click'))
    expect(screen.getByText('Click')).toBeInTheDocument()
    expect(snackRef.current?.show()).toBeUndefined()
  })
  it('should render correctly with buttonComponent Prop', () => {
    renderWithTheme(

      <Snackbar
        timer={5}
        buttonComponent={
          <Button onClick={() => ''}>Click</Button>
    }
      >
        {' '}
        Texto children

      </Snackbar>
    )
    expect(screen.getByTestId('btn-component')).toBeInTheDocument()
  })
  it('should render correctly with icon Prop', () => {
    renderWithTheme(

      <Snackbar
        timer={5}
        icon={<>Icon teste</>}
        buttonComponent={
          <Button onClick={() => ''}>Click</Button>
    }
      >
        {' '}
        Texto children

      </Snackbar>
    )
    expect(screen.getByTestId('icon-item')).toBeInTheDocument()
  })

  it('should render correctly with feedback success', () => {
    const { styles, component } = renderWithTheme(

      <Snackbar
        testID="bkg-color"
        timer={5}
        feedback="success"
      />

    )
    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should render correctly with feedback error', () => {
    const { styles, component } = renderWithTheme(

      <Snackbar
        timer={5}
        feedback="error"
      />

    )
    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly with feedback warning', () => {
    const { styles, component } = renderWithTheme(

      <Snackbar
        timer={5}
        feedback="warning"
      />

    )
    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly with feedback info', () => {
    const { styles, component } = renderWithTheme(

      <Snackbar
        timer={5}
        feedback="info"
      />

    )
    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
  it('should render correctly with feedback default', () => {
    const { styles, component } = renderWithTheme(

      <Snackbar
        timer={5}
        feedback="default"
      />

    )
    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
})
