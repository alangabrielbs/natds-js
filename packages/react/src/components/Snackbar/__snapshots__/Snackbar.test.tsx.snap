// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar component should render correctly with default props 1`] = `
Array [
  ".snackbarContainer-0-2-1 {
  color: #ffffff;
  width: 328px;
  z-index: 1000;
  position: fixed;
  flex-wrap: nowrap;
  font-size: 14px;
  min-height: 56px;
  align-items: center;
  font-family: Arial;
  font-weight: 400;
  white-space: nowrap;
  border-radius: 4px;
  justify-content: center;
}
.wrapper-0-2-2 {
  gap: 16px;
  width: 100%;
  display: flex;
  padding: 16px;
  overflow: hidden;
  box-sizing: border-box;
  align-items: center;
  justify-content: space-between;
}
.wrapperRow-0-2-3 {
  gap: 8px;
  width: 100%;
  display: flex;
  align-items: start;
}
.wrapperColumm-0-2-4 {
  gap: 8px;
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
}
.wrapperAction-0-2-5 {
  gap: 8px;
  color: #ffffff;
  display: flex;
  justify-content: end;
}
.bodyText-0-2-6 {
  display: flex;
  flex-wrap: wrap;
  box-sizing: border-box;
  word-break: break-all;
  white-space: normal;
}
.wrapperIcon-0-2-7 {
  display: flex;
  box-sizing: border-box;
}
.show-0-2-8 {
  visibility: visible;
  animation-fill-mode: forwards;
}
.close-0-2-9 {
  visibility: hidden;
  animation-fill-mode: forwards;
}
@keyframes keyframes-openSnackR-0-2-10 {
  from {
    right: -328px;
  }
  to {
    right: 8px;
  }
}
@keyframes keyframes-openSnackL-0-2-11 {
  from {
    left: -328px;
  }
  to {
    left: 8px;
  }
}
@keyframes keyframes-openSnackB-0-2-12 {
  from {
    bottom: -56px;
  }
  to {
    bottom: 8px;
  }
}
@keyframes keyframes-openSnackT-0-2-13 {
  from {
    top: -56px;
  }
  to {
    top: 8px;
  }
}
@keyframes keyframes-closeSnackR-0-2-14 {
  from {
    right: 8px;
  }
  to {
    right: -328px;
  }
}
@keyframes keyframes-closeSnackL-0-2-15 {
  from {
    left: 8px;
  }
  to {
    left: -328px;
  }
}
@keyframes keyframes-closeSnackB-0-2-16 {
  from {
    bottom: 8px;
  }
  to {
    bottom: -224px;
  }
}
@keyframes keyframes-closeSnackT-0-2-17 {
  from {
    top: 8px;
  }
  to {
    top: -224px;
  }
}
  .snackbarContainer-d0-0-2-18 {
    background-color: #000000;
    right: 8px ;
    top: 8px;
    bottom: unset;
    left: calc(50% - 164px);
  }
  .wrapper-d1-0-2-19 {
    flex-direction: row;
  }
  .wrapperAction-d2-0-2-20 {
    width: auto;
  }
  .show-d3-0-2-21 {
    animation: ;
  }",
  <div>
    <div
      class=" close-0-2-9 snackbarContainer-0-2-1 snackbarContainer-d0-0-2-18"
    >
      <div
        class="wrapper-0-2-2 wrapper-d1-0-2-19"
      >
        <div
          class="wrapperRow-0-2-3"
        >
          <div
            class="wrapperColumm-0-2-4"
            data-testid="title-item"
          >
            <div
              class="bodyText-0-2-6"
            />
          </div>
        </div>
        <div
          class="wrapperAction-0-2-5 wrapperAction-d2-0-2-20"
        >
          <span
            data-testid="btn-component"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Snackbar component should render correctly with feedback default 1`] = `
Array [
  ".snackbarContainer-0-2-1 {
  color: #ffffff;
  width: 328px;
  z-index: 1000;
  position: fixed;
  flex-wrap: nowrap;
  font-size: 14px;
  min-height: 56px;
  align-items: center;
  font-family: Arial;
  font-weight: 400;
  white-space: nowrap;
  border-radius: 4px;
  justify-content: center;
}
.wrapper-0-2-2 {
  gap: 16px;
  width: 100%;
  display: flex;
  padding: 16px;
  overflow: hidden;
  box-sizing: border-box;
  align-items: center;
  justify-content: space-between;
}
.wrapperRow-0-2-3 {
  gap: 8px;
  width: 100%;
  display: flex;
  align-items: start;
}
.wrapperColumm-0-2-4 {
  gap: 8px;
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
}
.wrapperAction-0-2-5 {
  gap: 8px;
  color: #ffffff;
  display: flex;
  justify-content: end;
}
.bodyText-0-2-6 {
  display: flex;
  flex-wrap: wrap;
  box-sizing: border-box;
  word-break: break-all;
  white-space: normal;
}
.wrapperIcon-0-2-7 {
  display: flex;
  box-sizing: border-box;
}
.show-0-2-8 {
  visibility: visible;
  animation-fill-mode: forwards;
}
.close-0-2-9 {
  visibility: hidden;
  animation-fill-mode: forwards;
}
@keyframes keyframes-openSnackR-0-2-10 {
  from {
    right: -328px;
  }
  to {
    right: 8px;
  }
}
@keyframes keyframes-openSnackL-0-2-11 {
  from {
    left: -328px;
  }
  to {
    left: 8px;
  }
}
@keyframes keyframes-openSnackB-0-2-12 {
  from {
    bottom: -56px;
  }
  to {
    bottom: 8px;
  }
}
@keyframes keyframes-openSnackT-0-2-13 {
  from {
    top: -56px;
  }
  to {
    top: 8px;
  }
}
@keyframes keyframes-closeSnackR-0-2-14 {
  from {
    right: 8px;
  }
  to {
    right: -328px;
  }
}
@keyframes keyframes-closeSnackL-0-2-15 {
  from {
    left: 8px;
  }
  to {
    left: -328px;
  }
}
@keyframes keyframes-closeSnackB-0-2-16 {
  from {
    bottom: 8px;
  }
  to {
    bottom: -224px;
  }
}
@keyframes keyframes-closeSnackT-0-2-17 {
  from {
    top: 8px;
  }
  to {
    top: -224px;
  }
}
  .snackbarContainer-d0-0-2-18 {
    background-color: #000000;
    right: 8px ;
    top: 8px;
    bottom: unset;
    left: calc(50% - 164px);
  }
  .wrapper-d1-0-2-19 {
    flex-direction: row;
  }
  .wrapperAction-d2-0-2-20 {
    width: auto;
  }
  .show-d3-0-2-21 {
    animation: ;
  }",
  <div>
    <div
      class=" close-0-2-9 snackbarContainer-0-2-1 snackbarContainer-d0-0-2-18"
    >
      <div
        class="wrapper-0-2-2 wrapper-d1-0-2-19"
      >
        <div
          class="wrapperRow-0-2-3"
        >
          <div
            class="wrapperColumm-0-2-4"
            data-testid="title-item"
          >
            <div
              class="bodyText-0-2-6"
            />
          </div>
        </div>
        <div
          class="wrapperAction-0-2-5 wrapperAction-d2-0-2-20"
        >
          <span
            data-testid="btn-component"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Snackbar component should render correctly with feedback error 1`] = `
Array [
  ".snackbarContainer-0-2-1 {
  color: #ffffff;
  width: 328px;
  z-index: 1000;
  position: fixed;
  flex-wrap: nowrap;
  font-size: 14px;
  min-height: 56px;
  align-items: center;
  font-family: Arial;
  font-weight: 400;
  white-space: nowrap;
  border-radius: 4px;
  justify-content: center;
}
.wrapper-0-2-2 {
  gap: 16px;
  width: 100%;
  display: flex;
  padding: 16px;
  overflow: hidden;
  box-sizing: border-box;
  align-items: center;
  justify-content: space-between;
}
.wrapperRow-0-2-3 {
  gap: 8px;
  width: 100%;
  display: flex;
  align-items: start;
}
.wrapperColumm-0-2-4 {
  gap: 8px;
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
}
.wrapperAction-0-2-5 {
  gap: 8px;
  color: #ffffff;
  display: flex;
  justify-content: end;
}
.bodyText-0-2-6 {
  display: flex;
  flex-wrap: wrap;
  box-sizing: border-box;
  word-break: break-all;
  white-space: normal;
}
.wrapperIcon-0-2-7 {
  display: flex;
  box-sizing: border-box;
}
.show-0-2-8 {
  visibility: visible;
  animation-fill-mode: forwards;
}
.close-0-2-9 {
  visibility: hidden;
  animation-fill-mode: forwards;
}
@keyframes keyframes-openSnackR-0-2-10 {
  from {
    right: -328px;
  }
  to {
    right: 8px;
  }
}
@keyframes keyframes-openSnackL-0-2-11 {
  from {
    left: -328px;
  }
  to {
    left: 8px;
  }
}
@keyframes keyframes-openSnackB-0-2-12 {
  from {
    bottom: -56px;
  }
  to {
    bottom: 8px;
  }
}
@keyframes keyframes-openSnackT-0-2-13 {
  from {
    top: -56px;
  }
  to {
    top: 8px;
  }
}
@keyframes keyframes-closeSnackR-0-2-14 {
  from {
    right: 8px;
  }
  to {
    right: -328px;
  }
}
@keyframes keyframes-closeSnackL-0-2-15 {
  from {
    left: 8px;
  }
  to {
    left: -328px;
  }
}
@keyframes keyframes-closeSnackB-0-2-16 {
  from {
    bottom: 8px;
  }
  to {
    bottom: -224px;
  }
}
@keyframes keyframes-closeSnackT-0-2-17 {
  from {
    top: 8px;
  }
  to {
    top: -224px;
  }
}
  .snackbarContainer-d0-0-2-18 {
    background-color: #e74627;
    right: 8px ;
    top: 8px;
    bottom: unset;
    left: calc(50% - 164px);
  }
  .wrapper-d1-0-2-19 {
    flex-direction: row;
  }
  .wrapperAction-d2-0-2-20 {
    width: auto;
  }
  .show-d3-0-2-21 {
    animation: ;
  }",
  <div>
    <div
      class=" close-0-2-9 snackbarContainer-0-2-1 snackbarContainer-d0-0-2-18"
    >
      <div
        class="wrapper-0-2-2 wrapper-d1-0-2-19"
      >
        <div
          class="wrapperRow-0-2-3"
        >
          <div
            class="wrapperColumm-0-2-4"
            data-testid="title-item"
          >
            <div
              class="bodyText-0-2-6"
            />
          </div>
        </div>
        <div
          class="wrapperAction-0-2-5 wrapperAction-d2-0-2-20"
        >
          <span
            data-testid="btn-component"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Snackbar component should render correctly with feedback info 1`] = `
Array [
  ".snackbarContainer-0-2-1 {
  color: #ffffff;
  width: 328px;
  z-index: 1000;
  position: fixed;
  flex-wrap: nowrap;
  font-size: 14px;
  min-height: 56px;
  align-items: center;
  font-family: Arial;
  font-weight: 400;
  white-space: nowrap;
  border-radius: 4px;
  justify-content: center;
}
.wrapper-0-2-2 {
  gap: 16px;
  width: 100%;
  display: flex;
  padding: 16px;
  overflow: hidden;
  box-sizing: border-box;
  align-items: center;
  justify-content: space-between;
}
.wrapperRow-0-2-3 {
  gap: 8px;
  width: 100%;
  display: flex;
  align-items: start;
}
.wrapperColumm-0-2-4 {
  gap: 8px;
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
}
.wrapperAction-0-2-5 {
  gap: 8px;
  color: #ffffff;
  display: flex;
  justify-content: end;
}
.bodyText-0-2-6 {
  display: flex;
  flex-wrap: wrap;
  box-sizing: border-box;
  word-break: break-all;
  white-space: normal;
}
.wrapperIcon-0-2-7 {
  display: flex;
  box-sizing: border-box;
}
.show-0-2-8 {
  visibility: visible;
  animation-fill-mode: forwards;
}
.close-0-2-9 {
  visibility: hidden;
  animation-fill-mode: forwards;
}
@keyframes keyframes-openSnackR-0-2-10 {
  from {
    right: -328px;
  }
  to {
    right: 8px;
  }
}
@keyframes keyframes-openSnackL-0-2-11 {
  from {
    left: -328px;
  }
  to {
    left: 8px;
  }
}
@keyframes keyframes-openSnackB-0-2-12 {
  from {
    bottom: -56px;
  }
  to {
    bottom: 8px;
  }
}
@keyframes keyframes-openSnackT-0-2-13 {
  from {
    top: -56px;
  }
  to {
    top: 8px;
  }
}
@keyframes keyframes-closeSnackR-0-2-14 {
  from {
    right: 8px;
  }
  to {
    right: -328px;
  }
}
@keyframes keyframes-closeSnackL-0-2-15 {
  from {
    left: 8px;
  }
  to {
    left: -328px;
  }
}
@keyframes keyframes-closeSnackB-0-2-16 {
  from {
    bottom: 8px;
  }
  to {
    bottom: -224px;
  }
}
@keyframes keyframes-closeSnackT-0-2-17 {
  from {
    top: 8px;
  }
  to {
    top: -224px;
  }
}
  .snackbarContainer-d0-0-2-18 {
    background-color: #227bbd;
    right: 8px ;
    top: 8px;
    bottom: unset;
    left: calc(50% - 164px);
  }
  .wrapper-d1-0-2-19 {
    flex-direction: row;
  }
  .wrapperAction-d2-0-2-20 {
    width: auto;
  }
  .show-d3-0-2-21 {
    animation: ;
  }",
  <div>
    <div
      class=" close-0-2-9 snackbarContainer-0-2-1 snackbarContainer-d0-0-2-18"
    >
      <div
        class="wrapper-0-2-2 wrapper-d1-0-2-19"
      >
        <div
          class="wrapperRow-0-2-3"
        >
          <div
            class="wrapperColumm-0-2-4"
            data-testid="title-item"
          >
            <div
              class="bodyText-0-2-6"
            />
          </div>
        </div>
        <div
          class="wrapperAction-0-2-5 wrapperAction-d2-0-2-20"
        >
          <span
            data-testid="btn-component"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Snackbar component should render correctly with feedback success 1`] = `
Array [
  ".snackbarContainer-0-2-1 {
  color: #ffffff;
  width: 328px;
  z-index: 1000;
  position: fixed;
  flex-wrap: nowrap;
  font-size: 14px;
  min-height: 56px;
  align-items: center;
  font-family: Arial;
  font-weight: 400;
  white-space: nowrap;
  border-radius: 4px;
  justify-content: center;
}
.wrapper-0-2-2 {
  gap: 16px;
  width: 100%;
  display: flex;
  padding: 16px;
  overflow: hidden;
  box-sizing: border-box;
  align-items: center;
  justify-content: space-between;
}
.wrapperRow-0-2-3 {
  gap: 8px;
  width: 100%;
  display: flex;
  align-items: start;
}
.wrapperColumm-0-2-4 {
  gap: 8px;
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
}
.wrapperAction-0-2-5 {
  gap: 8px;
  color: #ffffff;
  display: flex;
  justify-content: end;
}
.bodyText-0-2-6 {
  display: flex;
  flex-wrap: wrap;
  box-sizing: border-box;
  word-break: break-all;
  white-space: normal;
}
.wrapperIcon-0-2-7 {
  display: flex;
  box-sizing: border-box;
}
.show-0-2-8 {
  visibility: visible;
  animation-fill-mode: forwards;
}
.close-0-2-9 {
  visibility: hidden;
  animation-fill-mode: forwards;
}
@keyframes keyframes-openSnackR-0-2-10 {
  from {
    right: -328px;
  }
  to {
    right: 8px;
  }
}
@keyframes keyframes-openSnackL-0-2-11 {
  from {
    left: -328px;
  }
  to {
    left: 8px;
  }
}
@keyframes keyframes-openSnackB-0-2-12 {
  from {
    bottom: -56px;
  }
  to {
    bottom: 8px;
  }
}
@keyframes keyframes-openSnackT-0-2-13 {
  from {
    top: -56px;
  }
  to {
    top: 8px;
  }
}
@keyframes keyframes-closeSnackR-0-2-14 {
  from {
    right: 8px;
  }
  to {
    right: -328px;
  }
}
@keyframes keyframes-closeSnackL-0-2-15 {
  from {
    left: 8px;
  }
  to {
    left: -328px;
  }
}
@keyframes keyframes-closeSnackB-0-2-16 {
  from {
    bottom: 8px;
  }
  to {
    bottom: -224px;
  }
}
@keyframes keyframes-closeSnackT-0-2-17 {
  from {
    top: 8px;
  }
  to {
    top: -224px;
  }
}
  .snackbarContainer-d0-0-2-18 {
    background-color: #569a32;
    right: 8px ;
    top: 8px;
    bottom: unset;
    left: calc(50% - 164px);
  }
  .wrapper-d1-0-2-19 {
    flex-direction: row;
  }
  .wrapperAction-d2-0-2-20 {
    width: auto;
  }
  .show-d3-0-2-21 {
    animation: ;
  }",
  <div>
    <div
      class=" close-0-2-9 snackbarContainer-0-2-1 snackbarContainer-d0-0-2-18"
      data-testid="bkg-color"
    >
      <div
        class="wrapper-0-2-2 wrapper-d1-0-2-19"
      >
        <div
          class="wrapperRow-0-2-3"
        >
          <div
            class="wrapperColumm-0-2-4"
            data-testid="title-item"
          >
            <div
              class="bodyText-0-2-6"
            />
          </div>
        </div>
        <div
          class="wrapperAction-0-2-5 wrapperAction-d2-0-2-20"
        >
          <span
            data-testid="btn-component"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Snackbar component should render correctly with feedback warning 1`] = `
Array [
  ".snackbarContainer-0-2-1 {
  color: #ffffff;
  width: 328px;
  z-index: 1000;
  position: fixed;
  flex-wrap: nowrap;
  font-size: 14px;
  min-height: 56px;
  align-items: center;
  font-family: Arial;
  font-weight: 400;
  white-space: nowrap;
  border-radius: 4px;
  justify-content: center;
}
.wrapper-0-2-2 {
  gap: 16px;
  width: 100%;
  display: flex;
  padding: 16px;
  overflow: hidden;
  box-sizing: border-box;
  align-items: center;
  justify-content: space-between;
}
.wrapperRow-0-2-3 {
  gap: 8px;
  width: 100%;
  display: flex;
  align-items: start;
}
.wrapperColumm-0-2-4 {
  gap: 8px;
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
}
.wrapperAction-0-2-5 {
  gap: 8px;
  color: #ffffff;
  display: flex;
  justify-content: end;
}
.bodyText-0-2-6 {
  display: flex;
  flex-wrap: wrap;
  box-sizing: border-box;
  word-break: break-all;
  white-space: normal;
}
.wrapperIcon-0-2-7 {
  display: flex;
  box-sizing: border-box;
}
.show-0-2-8 {
  visibility: visible;
  animation-fill-mode: forwards;
}
.close-0-2-9 {
  visibility: hidden;
  animation-fill-mode: forwards;
}
@keyframes keyframes-openSnackR-0-2-10 {
  from {
    right: -328px;
  }
  to {
    right: 8px;
  }
}
@keyframes keyframes-openSnackL-0-2-11 {
  from {
    left: -328px;
  }
  to {
    left: 8px;
  }
}
@keyframes keyframes-openSnackB-0-2-12 {
  from {
    bottom: -56px;
  }
  to {
    bottom: 8px;
  }
}
@keyframes keyframes-openSnackT-0-2-13 {
  from {
    top: -56px;
  }
  to {
    top: 8px;
  }
}
@keyframes keyframes-closeSnackR-0-2-14 {
  from {
    right: 8px;
  }
  to {
    right: -328px;
  }
}
@keyframes keyframes-closeSnackL-0-2-15 {
  from {
    left: 8px;
  }
  to {
    left: -328px;
  }
}
@keyframes keyframes-closeSnackB-0-2-16 {
  from {
    bottom: 8px;
  }
  to {
    bottom: -224px;
  }
}
@keyframes keyframes-closeSnackT-0-2-17 {
  from {
    top: 8px;
  }
  to {
    top: -224px;
  }
}
  .snackbarContainer-d0-0-2-18 {
    background-color: #fcc433;
    right: 8px ;
    top: 8px;
    bottom: unset;
    left: calc(50% - 164px);
  }
  .wrapper-d1-0-2-19 {
    flex-direction: row;
  }
  .wrapperAction-d2-0-2-20 {
    width: auto;
  }
  .show-d3-0-2-21 {
    animation: ;
  }",
  <div>
    <div
      class=" close-0-2-9 snackbarContainer-0-2-1 snackbarContainer-d0-0-2-18"
    >
      <div
        class="wrapper-0-2-2 wrapper-d1-0-2-19"
      >
        <div
          class="wrapperRow-0-2-3"
        >
          <div
            class="wrapperColumm-0-2-4"
            data-testid="title-item"
          >
            <div
              class="bodyText-0-2-6"
            />
          </div>
        </div>
        <div
          class="wrapperAction-0-2-5 wrapperAction-d2-0-2-20"
        >
          <span
            data-testid="btn-component"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;
